File: 1129/02/snippet.cpp
#include <iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main() {
    string a, b;
    cin >> a >> b;
    int total_minute = 24 * 60, mint = 1e+6, maxt = 0;
    vector<int> vec1, vec2;
    for (int i = 0; i < total_minute; ++i) {
        int hour = i / 60, minute = i % 60;
        if ((a[0] == '?' || a[0] - '0' == hour / 10) && (a[1] == '?' ||
                a[1] - '0' == hour % 10) && (a[3] == '?' || a[3] - '0' == minute / 10) &&
                (a[4] == '?' || a[4] - '0' == minute % 10))
            vec1.push_back(i);
        if ((b[0] == '?' || b[0] - '0' == hour / 10) && (b[1] == '?' ||
                b[1] - '0' == hour % 10) && (b[3] == '?' || b[3] - '0' == minute / 10) &&
                (b[4] == '?' || b[4] - '0' == minute % 10))
            vec2.push_back(i);
    }
    for (int i = 0; i < vec1.size(); ++i) {
        for (int j = 0; j < vec2.size(); ++j) {
            if (vec1[i] < vec2[j]) {
                mint = min(vec2[j] - vec1[i], mint);
                maxt = max(vec2[j] - vec1[i], maxt);
            }
        }
    }
    cout << mint << " " << maxt;
}
// 64 位输出请用 printf("%lld")

File: 1129/02/snippet.py
import sys

s1 = input()
s2 = input()

list1 = []
list2 = []

for i in range(0, 60 * 24):
    hour = int(i // 60)
    minute = int(i % 60)

    if s1[0] == '?' or int(s1[0]) == hour // 10:
        if s1[1] == '?' or int(s1[1]) == hour % 10:
            if s1[3] == '?' or int(s1[3]) == minute // 10:
                if s1[4] == '?' or int(s1[4]) == minute % 10:
                    list1.append(i)

    if s2[0] == '?' or int(s2[0]) == hour // 10:
        if s2[1] == '?' or int(s2[1]) == hour % 10:
            if s2[3] == '?' or int(s2[3]) == minute // 10:
                if s2[4] == '?' or int(s2[4]) == minute % 10:
                    list2.append(i)

mintime = 60*24
maxtime = 0
for i in range(len(list1)):
    for j in range(len(list2)):
        if list1[i] < list2[j]:
            mintime = min(mintime, list2[j] - list1[i])
            maxtime = max(maxtime, list2[j] - list1[i])

print(mintime, maxtime)